### ISSUE TITLE
[testing] SkipInfo should error if `cls_name` and `test_name` combination is not valid

### ISSUE THREAD

**Issue by kshitij12345** [pytorch/pytorch#60223]
Currently `SkipInfo` does nothing if `cls_name` and `test_name` don't match any expected test.

However for this to work `SkipInfo` will **probably** (not sure about the implementation) have to know all the `Test` class and `test names` (not sure if this is possible) but will be nice if it can error out.

cc: @mruberry

**Comment by mruberry** [comment]
I agree, but I also couldn't think of a way to fix this. Leaving it open in case we discover a mechanism to enforce this property.
